src/ops/ops.go

```diff
@@ -463,3 +465,44 @@ func (o *ops) CreateOpenshiftSshManifest(filePath, tmpl, sshPubKeyPath string) e
 	}
 	return nil
 }
+
+func (o *ops) GetMustGatherLogs(workDir string, kubeconfigPath string) (string, error) {
+	//invoke oc adm must-gather command in the working directory
+	o.log.Infof("collecting must-gather logs into %s", workDir)
+	command := fmt.Sprintf("cd %s && oc --kubeconfig=%s adm must-gather", workDir, kubeconfigPath)
+	output, err := o.ExecCommand(o.logWriter, "bash", "-c", command)
+	if err != nil {
+		return "", err
+	}
+	o.log.Info(output)
+
+	//find the directory of logs which is the output of the command
+	//this is a temp directory so we have to find it by its prefix
+	files, err := ioutil.ReadDir(workDir)
```

Comment:
can't we have this directory as not tmp? We will run this code one in the container life so not sure why it worth to have it as tmp with unique name

Reply:
maximum we can delete if exists

Reply:
I use tmp directory for safety reasons. all the files related to the process are located there and not in the home directory and than it can be released. the embedded directory is related to how the command works. I truly think its for the best

Reply:
but we have 4 times must-gather in different places

Reply:
7 times

Reply:
not sure it worth it. U can have just hard coded folder name and u will not need all search operation

Reply:
The search code is not for my directory but for the directory generated by the oc must-gather command. They generate directory name with rand suffix so I am obliged to search for it by its fixed prefix anyhow, no matter what my parent directory looks like (temp or not)